cli/src/git_util.rs

```diff
@@ -275,6 +275,24 @@ pub fn with_remote_git_callbacks<T>(ui: &Ui, f: impl FnOnce(git::RemoteCallbacks
         |url: &str| Some((terminal_get_username(ui, url)?, terminal_get_pw(ui, url)?));
     callbacks.get_username_password = Some(&mut get_user_pw);
 
+    #[cfg(feature = "git2")]
+    let mut git2_deprecation_warning;
+    #[cfg(feature = "git2")]
+    if std::env::var("JJ_DEBUG_SUPPRESS_GIT2_DEPRECATION_WARNING").as_deref() != Ok("1") {
+        use std::sync::Once;
+        static GIT2_WARNING: Once = Once::new();
+        git2_deprecation_warning = || {
+            GIT2_WARNING.call_once(|| {
+                _ = writeln!(
+                    ui.warning_default(),
+                    "`git.subprocess = false` will be removed in 0.30; please report any issues \
+                     you have with the default.",
+                );
+            });
+        };
+        callbacks.git2_deprecation_warning = Some(&mut git2_deprecation_warning);
+    }
```

Comment:
I don't think we need callback. Can't we instead add `warn(ui, &git_settings)` function and let `cmd_git_*()` functions call it?

Maybe we can also use `debug.suppress-git2-deprecation-warning` config key instead (because `UserSettings` should be available there.)

Reply:
Yeah, that works, and produces slightly nicer output. Done. (This also means that `with_git_subprocess` could have been a regular method after all, but it doesn’t feel worth the churn to change it.)